CMakeLists.txt

```diff
@@ -80,7 +80,11 @@ IF(NOT KOKKOS_HAS_TRILINOS)
     # WORKAROUND FOR HIPCC
     IF(Kokkos_ENABLE_HIP)
       SET(KOKKOS_INTERNAL_CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
-      SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --amdgpu-target=gfx906")
+      IF("${KOKKOS_CXX_COMPILER_ID}" STREQUAL Clang)
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --cuda-host-only")
```

Comment:
> Fixup CMake to detect HIP-Clang vs HCC, and use the `--cuda-host-only` for GTest compilation.

Can you elaborate on this? You need to disable the `CUDA` pass in `clang` for being able to compile for `AMD` GPUs?

Reply:
This is a case of CUDA-Clang's naming being all over the upstream compiler -- we piggy back off [cuda-host-only](https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-cuda-host-only) to perform the same thing for AMD GPUs.  Really it should be `--gpu-host-only` or something, but that name seems fraught as well :)